integr8ly/cloud-resource-operator
# MGDAPI-3258 - Update Go dependency to 1.17

## Overview

Jira: [MGDAPI-3258](https://issues.redhat.com/browse/MGDAPI-3258)

## Changes
* Updated Go version for CRO to 1.17
* Updated `controller-gen` from `v0.3.0` to `v0.8.0` (Go  version 1.13 -> 1.17)
* Updated `kustomize` from `v3@3.5.4` to `v4@v4.4.1` (Go version 1.13 -> 1.16)

## Verification
* Check `make code/gen` works as expected with new controller-gen version

Ensure main functions of CRO are unaffected, on CCS cluster:
* `make cluster/prepare`
* `make run`
* Deploy `Postgres`, `BlobStorage`, `Redis` CRs
* Delete `Postgres`, `BlobStorage`, `Redis` CRs
(Check AWS resources to ensure this completes as expected)

## Checklist
